packages/common/src/types.ts

```diff
@@ -108,26 +108,138 @@ export type IBatch = any[];
 export type IBatchTransform = (value: any) => any;
 
 export interface IGetSurveyModelsResponse {
-  form: IModel,
-  featureService: IModel,
-  fieldworker: IModel,
-  stakeholder: IModel
-};
+  form: IModel;
+  featureService: IModel;
+  fieldworker: IModel;
+  stakeholder: IModel;
+}
 
 export interface IGetGroupSharingDetailsResults {
-  group: IGroup,
-  modelsToShare: IModel[]
-};
+  group: IGroup;
+  modelsToShare: IModel[];
+}
 
 export interface IRevertableTaskSuccess {
-  status: "fullfilled",
-  revert: (...args: any[]) => Promise<any>,
-  results: any
-};
+  status: "fullfilled";
+  revert: (...args: any[]) => Promise<any>;
+  results: any;
+}
 
 export interface IRevertableTaskFailed {
-  status: "rejected",
-  error: Error
-};
+  status: "rejected";
+  error: Error;
+}
+
+export type IRevertableTaskResult =
+  | IRevertableTaskSuccess
+  | IRevertableTaskFailed;
+
+/**
+ * Hub Types
+ */
+export type HubType =
+  | "member"
+  | "team"
+  | "event"
+  | "dataset"
+  | "document"
+  | "map "
+  | "app "
+  | "site"
+  | "initiative"
+  | "template"
+  | "organization";
+export type VisibilityOptions = "private" | "org" | "public";
+export type ControlOptions = "view" | "edit" | "admin";
+
+// TW: basing this on https://opendata.arcgis.com/api/v3/datasets/7a153563b0c74f7eb2b3eae8a66f2fbb_0/
+// but that may not be accurate
+export interface IHubGeography {
+  size?: number; // what is this?
+  center?: [number, number];
+  geometry?: IGeometry;
+}
+
+export interface IHubResource {
+  name: string; // Generic term for the primary label (title, fullname, username, etc.)
+  summary?: string; // snippet or other summary
+  // publisher: IHubOwner // TODO: better name? item.owner with more user metadata
+
+  // Derived metadata
+  hubType: HubType;
+  // TW: why is this on IHubResource instead of IHubContent?
+  permissions: {
+    // overrides item.access with more attributes. could flatten.
+    visibility: VisibilityOptions; // item.access
+    control?: ControlOptions; // itemControl
+    groups?: IGroup[]; // item.sharing.groups via content/users/:username/items/:id
+  };
 
-export type IRevertableTaskResult = IRevertableTaskSuccess | IRevertableTaskFailed;
+  // Explicit data information since this is a common confusion + bug report
+  createdDate: Date; // formal metadata || new Date(item.created)
+  createdDateSource?: string; // description of what was used for this attribute
```

Comment:
comment may be explicit that this is meant to be the attribute key, e.g. `item.modified` or `item.metadata.created_date` which would likely match to a UI string key.